rpc/provider/smoldot.ts

```diff
@@ -67,27 +91,38 @@ async function connection(
   }
   let conn = connections.get(chainSpec);
   if (!conn) {
-    // TODO: try catch this and send through handler within a `ProviderHandlerError`
-    const inner = await client.addChain({ chainSpec });
-    conn = new SmoldotProviderConnection(inner, () => {
-      try {
-        inner.remove();
-      } catch (_e) { /* TODO */ }
-    });
+    let inner: Chain;
+    if (typeof chainSpec === "string") {
+      inner = await client.addChain({ chainSpec });
+    } else {
+      const [parachainSpec, relayChainSpec] = chainSpec;
+      const relayChainConnection = await client.addChain({ chainSpec: relayChainSpec });
+      inner = await client.addChain({
+        chainSpec: parachainSpec,
+        potentialRelayChains: [relayChainConnection],
+      });
+    }
```

Comment:
@harrysolovay @wirednkod WDYT?

Reply:
Giving my 2 cents here, I think this will strict the user always to start relay chain and parachain at the same spot of the application... IMO these steps should be separated (add relay chain, add parachain) with the same function

Reply:
thanks for the input @wirednkod 
For now, the relay/parachain start will be on demand on the first `.send` invocation.